reframe/core/containers.py

```diff
@@ -119,32 +116,45 @@ class Sarus(ContainerPlatform):
     #: :default: :class:`False`
     with_mpi = fields.TypedField(bool)
 
+    #: Skip pull of images
+    #:
+    #: :type: boolean
+    #: :default: :class:`False`
+    skip_pull = fields.TypedField(bool)
+
     def __init__(self):
         super().__init__()
         self.with_mpi = False
         self._command = 'sarus'
+        self.skip_pull = False
 
     def emit_prepare_commands(self):
         # The format that Sarus uses to call the images is
         # <reposerver>/<user>/<image>:<tag>. If an image was loaded
         # locally from a tar file, the <reposerver> is 'load'.
         if self.image.startswith('load/'):
             return []
+        # For testing purposes a way to skip the pull of an image.
+        if self.skip_pull:
+            return []
 
         return [self._command + ' pull %s' % self.image]
 
     def launch_command(self):
         super().launch_command()
         run_opts = ['--mount=type=bind,source="%s",destination="%s"' %
                     mp for mp in self.mount_points]
+        run_opts.append('--workdir='+self.workdir)
         if self.with_mpi:
             run_opts.append('--mpi')
 
         run_opts += self.options
-        run_cmd = self._command + ' run %s %s bash -c ' % (' '.join(run_opts),
+        run_cmd = self._command + ' run %s %s' % (' '.join(run_opts),
```

Comment:
Again, the entrypoint you specify here (`bash -c`) might mess with the container image and renders it unusable.